app/services/reservation_creator.rb

```diff
@@ -104,4 +107,19 @@ def status_q
     ActiveSupport::StringInquirer.new(@success.to_s)
   end
 
+  def purchase_original_cross_core_order
+    original_order = Order.find(@order.merge_with_order_id)
+
+    if original_order.order_details.one? && original_order.cross_core_project.present?
+      order_purchaser = OrderPurchaser.new(
+        order: original_order,
+        acting_as: original_order.user,
+        order_in_past: nil,
+        params: ActionController::Parameters.new({}),
+        user: original_order.created_by
+      )
+      order_purchaser.purchase_cross_core!
```

Comment:
What happens if this line fails?

Reply:
<img width="1189" alt="image" src="https://github.com/tablexi/nucore-open/assets/28798610/d7413ac4-069c-45f3-815a-a106cf38711d">